lib/models/task.dart

```diff
@@ -160,27 +158,27 @@ class Task {
   }) {
     return Task(
       id: id ?? this.id,
-      parentTaskId: parentTaskId,
-      priority: priority,
-      listId: listId,
-      bucketId: bucketId,
-      created: created,
-      updated: updated,
-      dueDate: dueDate,
-      startDate: startDate,
-      endDate: endDate,
-      reminderDates: reminderDates,
-      title: title,
-      description: description,
+      parentTaskId: parentTaskId ?? this.parentTaskId,
+      priority: priority ?? this.priority,
+      listId: listId ?? this.listId,
+      bucketId: bucketId ?? this.bucketId,
+      created: created ?? this.created,
+      updated: updated ?? this.updated,
+      dueDate: dueDate ?? this.dueDate,
+      startDate: startDate ?? this.startDate,
+      endDate: endDate ?? this.endDate,
+      reminderDates: reminderDates ?? this.reminderDates,
+      title: title ?? this.title,
+      description: description ?? this.description,
       identifier: identifier ?? this.identifier,
       done: done ?? this.done,
-      color: (resetColor ?? false) ? vBlue : (color ?? this.color),
-      kanbanPosition: kanbanPosition,
-      createdBy: createdBy,
-      repeatAfter: repeatAfter,
-      subtasks: subtasks,
-      labels: labels,
-      attachments: attachments,
+      color: (resetColor ?? false) ? null : (color ?? this.color),
+      kanbanPosition: kanbanPosition ?? this.kanbanPosition,
+      createdBy: createdBy ?? this.createdBy,
+      repeatAfter: repeatAfter ?? this.repeatAfter,
+      subtasks: subtasks ?? this.subtasks,
+      labels: labels ?? this.labels,
+      attachments: attachments ?? this.attachments,
```

Comment:
The idea of the `copyWith` method is that it returns a copy of the Task with the modifications specified.